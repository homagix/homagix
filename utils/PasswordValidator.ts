export function isValidPassword(password: string) {
  return password.length >= 8
}
