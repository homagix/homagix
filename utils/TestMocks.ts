let cookie: unknown

export function useMockedCookie() {
  return {
    get value() {
      return cookie
    },
    set value(val: unknown) {
      cookie = val
    },
  }
}
