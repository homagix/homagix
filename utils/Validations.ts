export function isValidPassword(password: string) {
  return password.length >= 8
}

export function isValidURL(url: string) {
  return url?.match(/^(https?:\/\/)?([a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)+)(:[0-9]{1,5})?(\/[^\s]*)?$/) !== null
}
