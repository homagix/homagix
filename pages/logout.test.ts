import { afterEach, describe, expect, it, vi } from "vitest"
import LogoutPage from "./logout.vue"
import { mockNuxtImport } from "@nuxt/test-utils/runtime"
import { setup } from "@nuxt/test-utils"
import { mount } from "@vue/test-utils"
import { useMockedCookie } from "~/utils/TestMocks"

const push = vi.fn()

describe("LogoutPage component", () => {
  setup({ server: false })

  afterEach(() => vi.restoreAllMocks())

  it("renders the title and description", async () => {
    const wrapper = mount(LogoutPage)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("sets the token cookie to null", async () => {
    mockNuxtImport("useCookie", () => () => useMockedCookie())
    const cookie = useCookie("token")
    cookie.value = "test-cookie"
    mount(LogoutPage)
    expect(cookie.value).toEqual(null)
  })

  it('navigates to "/" when the button is clicked', async () => {
    mockNuxtImport("useRouter", () => vi.fn(() => ({ push })))
    const wrapper = mount(LogoutPage)
    const button = wrapper.find("#recipes-button")
    await button.trigger("click")

    expect(push).toHaveBeenCalledWith("/")
  })
})
