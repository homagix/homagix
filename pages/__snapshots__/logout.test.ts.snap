// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LogoutPage component > renders the title and description 1`] = `
"<h2>Ciao!</h2>
<p>Wir hoffen, Dich bald wiederzusehen!</p>
<div class="button-list"><button id="recipes-button">Zu den Rezepten</button></div>"
`;
